library/core/src/ptr/const_ptr.rs

```diff
@@ -1347,16 +1366,26 @@ impl<T: ?Sized> *const T {
     /// # Panics
     ///
     /// The function panics if `align` is not a power-of-two (this includes 0).
+    // #[cfg(not(bootstrap))] -- Calling this function in a const context from the bootstrap
+    // compiler will always return false.
     #[must_use]
     #[inline]
     #[unstable(feature = "pointer_is_aligned", issue = "96284")]
-    pub fn is_aligned_to(self, align: usize) -> bool {
-        if !align.is_power_of_two() {
-            panic!("is_aligned_to: align is not a power-of-two");
+    #[rustc_const_unstable(feature = "const_pointer_is_aligned", issue = "none")]
+    pub const fn is_aligned_to(self, align: usize) -> bool {
+        assert!(align.is_power_of_two(), "is_aligned_to: align is not a power-of-two");
```

Comment:
This will be a slightly uglier panic message than before, since it will also print the stringified expression.

Reply:
There doesn't seem to be a (significant) difference to me, but I've changed it back for now. [(Goldbolt diff)](https://rust.godbolt.org/z/Pqdvzj5vY)